Dockerfile/bitcoind/Dockerfile

```diff
@@ -80,4 +80,4 @@ RUN groupadd -r bitcoin && useradd -r -m -g bitcoin bitcoin \
 
 EXPOSE 8332 8333 18332 18333 18443 18444
 
-ENTRYPOINT ["/bin/bash", "-c", "bitcoind -txindex -datadir=$BITCOIN_DATA"]
+ENTRYPOINT ["bitcoind", "-txindex", "-datadir=/var/lib/bitcoind"]
```

Comment:
because the entrypoint syntax we are using (exec form) doesn't expand env variables (unless you launch the comand with bash, as before).

the shell form of the command would be: `ENTRYPOINT bitcoind -txindex -datadir=$BITCOIN_DATA`, but this form has several drawbacks so I won't suggest to use that (https://docs.docker.com/engine/reference/builder/#entrypoint)

I had to remove `bash -c` to allow adding flags to the entrypoint via the command option

Reply:
> I had to remove bash -c to allow adding flags to the entrypoint via the command option

Strange, these options https://github.com/LNP-BP/docker/blob/master/docker-compose/mainnet/docker-compose.yml#L7-L8 were working with shell w/o any problems. I do not think that adding `-testnet` to them will render them not working...

What I'd like to achieve is not to (a) diverge different docker-compose and Dockerfiles between networks + (b) use ENV variables as much as possible

Reply:
I'll try to explain better:

using this entrypoint: `ENTRYPOINT ["/bin/bash", "-c", "bitcoind -txindex -datadir=$BITCOIN_DATA"]` the flags passed via the `command` option in the compose file were not added correctly to the command

using this entrypoint: ` ENTRYPOINT ["bitcoind", "-txindex", "-datadir=/var/lib/bitcoind"]` command options are correctly added to the entrypoint command, but this syntax doesn't allow env variables like `$BITCOIN_DATA`

it's not adding `-testnet` that makes `command` options not working.

another problem was with the list form:
```
command:
      - "-rpcuser=bitcoin -rpcpassword=bitcoin"
```
with this form, when I tried to perform a call to bitcoind (from the container itself) it was telling me that the credentials were wrong. When I switched to this form:
```
command: "-rpcuser=bitcoin -rpcpassword=bitcoin"
```
the internal calls started to work. So I suppose the options were just ignored, do you have any documentation about the list form being supported by docker-compose?

I agree with your requirements, but being mainnet the default network of pretty much everything, some dockerfiles changes were needed in order to let them work also on testnet.

Reply:
Thank you for the detailed explanation. I do not get what happens with this `command` thing; I clearly can connect to the bitcoind with bitcoin-cli set up with the same user and password...

Reply:
UPD: see my complete logs in the last comments, may be they can help to figure out what's going on

Reply:
Ok, I think I am getting what happens with env variable: it is expanded during `Dockerfile` build and within the image we don't have it anymore. That's why it is "working" even when no `bash` is used in `ENTRYPOINT`. To get it working like a real environment variable we need to use `${BITCOIN_DATA}` form, not just `$BITCOIN_DATA`.

However here `BITCOIN_DATA` is used for a single purpose: make sure that all paths within `Dockerfile` are the same. So we do not need it to expand and you can safely leave it here even when no bash is used.

The `-rpcpath` compose command is another story, which I will research later.

Reply:
Pls disregard my previous comment, I was wrong: https://docs.docker.com/engine/reference/builder/#cmd